mariadb_config/mariadb_config.c.in

```diff
@@ -15,6 +15,7 @@ static char *mariadb_progname;
 #define SOCKET  "@MARIADB_UNIX_ADDR@"
 #define PORT "@MARIADB_PORT@"
 #define TLS_LIBRARY_VERSION "@TLS_LIBRARY_VERSION@"
+#define EMBED_LIBS "-L@CMAKE_INSTALL_PREFIX@/@INSTALL_LIBDIR@/ -lmariadbd @extra_dynamic_LDFLAGS@"
```

Comment:
This should only work, if Connector/C is build within server and embedded option was set (default is off), so you should put the embedded stuff between 
```
#ifdef HAVE_EMBEDDED
..
#endif
```

In mariadb_config/CMakeLists.txt you have to check the build type:
```
IF(IS_SUBPROJECT AND WITH_EMBEDDED_SERVER)
  ADD_DEFINITIONS(-DHAVE_EMBEDDED)
ENDIF()
```